### DISCUSSION TITLE
prevent setting focus on closing a p:dialog

### DISCUSSION THREAD

**Discussion by hgfrey** [primefaces/community#614] [PrimeFaces] [answered]
How can I prevent that when closing a p:dialog the focus is set to the element that had focus before the dialog was opened? Or can I explicitly specify which element should receive focus? The problem is that the button that was previously pressed is no longer there after closing the dialog.

Thank you

**Comment by melloware** [comment]
@hgfrey this is typically and ARIA requirement for focusable dialogs.  Are you getting a JS Console error that the element no longer exists or just that nothing is focused after closing the dialog?

**Comment by hgfrey** [comment]
Yes, I get an error message in the JS Console: 'Widget for var 'widget_mainContentFormBerechtigung_table_anBerechtigung_0_btn_cancel_an' not available!' This must bee a generated message by PF.

![widget not availabe](https://github.com/primefaces/community/assets/33621098/353533da-79b3-42f7-a1ce-9764e2aed279)

What I find strange is that the pressed button (the one with the X for cancel) still seems to have focus. However, calling document.activeElement on the JS Console tells me that the Yes button has the input focus. I have other cases that are very similar. A row of data is to be created and the user presses cancel. Everything works perfectly and the cancel button is not selected in the background. So it doesn't seem that the default behavior is to blame. But I can't find any difference in my pages.

What I also find strange is that a missing widget is being complained about here. For a cancel button? What's that supposed to mean?

**Reply by melloware** [reply]
Can you provide an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project so I can debug your issue?

**Reply by hgfrey** [reply]
Unfortunately I am still unable to download the Jetty plugin for maven. There are things that are not so easy to do in a large company. I'll ask again tomorrow, but I'll try to strip down my sources.

**Reply by melloware** [reply]
Yeah if you can strip your sources down to a XTHML page snippet that I can reproduce your issue would be great.

**Comment by hgfrey** [comment, marked as answer]
I found the mistake. I need to trigger an update on the @Form. This causes the Cancel button to lose focus.
